### PR TITLE
Source ClickupApi: Add streams `time_tracking`, `time_tracking_tags`, `team_goals`, `space_tags`, `team_custom_fields`, `list_custom_fields`, `list_comments`, Add error handlers, Parent stream flow correction, Update spec

### LINKED ISSUE #44178: ClickUp source: add `Time Tracking` stream
## Connector Name
source-clickup

## ❓ What
The ClickUp connector does not include Time Tracking information, which retrieves information about time entries.

## 📈  Impact
Users can use the time entry information in ClickUp

## 🤝 Proposed solution
Add a new stream `time_tracking` using the endpoint: hhttps://clickup.com/api/clickupreference/operation/Gettimeentryhistory/
(Optional) Add a new stream `time_tracking_tags` using the endpoint: https://clickup.com/api/clickupreference/operation/Getalltagsfromtimeentries/

```[tasklist]
### Tasks
- [x] Source ClickupApi: Add streams `time_tracking`, `time_tracking_tags`, `team_goals`, `space_tags`, `team_custom_fields`, `list_custom_fields`, `list_comments`
- [x] Source ClickupApi: Add error handlers
- [x] Source ClickupApi: Fix parent child flow
```

### PR DESCRIPTION
**Pull request by btkcodedev**
## What
Closes https://github.com/airbytehq/airbyte/issues/44178

## How
- [x] Add streams `time_tracking`, `time_tracking_tags`, `team_goals`, `space_tags`, `team_custom_fields`, `list_custom_fields`, `list_comments`
- [x] Add error handlers
- [x] Fix parent child flow
- [x] Update docs

[screen-capture (1).webm](https://github.com/user-attachments/assets/42efa930-2d91-4119-9b7e-8e1f15c170f6)

### CODE REVIEW

**Comment by btkcodedev** [comment]
Tested & verified with builder :heavy_check_mark:

File: airbyte-integrations/connectors/source-clickup-api/metadata.yaml
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-clickup-api/metadata.yaml:8]: Lol @ChristoGrab @btkcodedev is going to beat your manifest-only migration ;D

File: airbyte-integrations/connectors/source-clickup-api/source_clickup_api/manifest.yaml
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-clickup-api/source_clickup_api/manifest.yaml:112]: @btkcodedev why ignore 404s? How are you using these, and where do they show up? Usually 404 could be a sign that you're trying to fetch a page that doesn't exist or something?

File: airbyte-integrations/connectors/source-clickup-api/source_clickup_api/manifest.yaml
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-clickup-api/source_clickup_api/manifest.yaml:103]: 500 should probably be a retry with backoff, no? I'm not an expert on our error handlers, what do @pnilan and @ChristoGrab think?

File: airbyte-integrations/connectors/source-clickup-api/source_clickup_api/manifest.yaml
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-clickup-api/source_clickup_api/manifest.yaml:740]: These blocks are populated when you do test reads in Builder, and we want to see them in most contributions. Good work @btkcodedev

Comment (natikgadzhi) [reply airbyte-integrations/connectors/source-clickup-api/source_clickup_api/manifest.yaml:740]: /cc @lmossman it works!

**Review by natikgadzhi — APPROVED** [review]
Approving to unblock. I'm not opposed to these, but I have questions re: error handlers.

/cc @nataliekwong rejoice, we're adding streams now!

**Comment by natikgadzhi** [comment]
Note to self: before merging I want to look at the regression test report and see how we ran it and what we have there.

**Comment by nataliekwong** [comment]
Epic!!

**Comment by btkcodedev** [comment]
@natikgadzhi  Conflicts cleared and CI is green again :bowing_man:

**Pull request merged**
